overlord/snapshotstate/backend/backend.go

```diff
@@ -486,34 +498,48 @@ var ErrCannotCancel = errors.New("cannot cancel: import already finished")
 // a commit.
 type importTransaction struct {
 	id        uint64
+	flockPath string
+	flock     *osutil.FileLock
```

Comment:
can you explain why you switch to have file locks? because of more than one snapd could be running at  the same time? or something else?

Reply:
Maybe it's not needed. My thinking was mostly that if something calls "snapshotstate.Import()" while backend.CleanupAbandondedImports() is in progress things would be bad. I don't think this can happen today because we do the cleanup in the StartUp() code so the daemon will not reply to API yet (AIUI). So maybe YAGNI and I could remove the flock. OTOH we still need to keep marker files around so it's not a huge win if we remove the locks (but happy to do it).

Reply:
The other advantage is that we could call CleanupAbandondedImports periodically - but again, if the exiting code does not have bugs it will never have to cleanup anything so probably not a good reason.

Reply:
I'm not against haveing the locks but as you said that there not clearly currently needed. I asked because we never use file locks far except when interfacting with other procesees. I'm fine if you want to keep them but they probably need a comment like:

```
// flock is a file lock associated with the import, the logic using it prevents potentially cleaning up/cancelling a still in
// progress import in case we started allowing cleaning and imports to happen concurrently
```